Sources/eppo/SdkTokenDecoder.swift

```diff
@@ -0,0 +1,66 @@
+import Foundation
+
+/// Decodes SDK tokens with embedded encoded data.
+public class SdkTokenDecoder {
+    private let decodedParams: [String: String]?
+    private let sdkToken: String
+    
+    public init(_ sdkToken: String) {
+        self.decodedParams = SdkTokenDecoder.decodeToken(sdkToken)
+        self.sdkToken = sdkToken
+    }
+    
+    /// Checks if the token is valid and contains encoded data.
+    public func isValid() -> Bool {
+        decodedParams != nil
+    }
+    
+    /// Returns the original token string.
+    public func getToken() -> String {
+        sdkToken
+    }
+    
+    /// Gets the subdomain from the token if available.
+    /// Returns nil if the token is invalid or doesn't contain a subdomain.
+    public func getSubdomain() -> String? {
+        decodedParams?["cs"]
+    }
+    
+    /// Gets the event ingestion hostname from the token if available.
+    /// Returns nil if the token is invalid or doesn't contain an event hostname.
+    public func getEventIngestionHostname() -> String? {
+        decodedParams?["eh"]
+    }
```

Comment:
event ingestion was never supported on iOS - an argument can be made that we can jump directly over the `eh` hostname and "keep it simple". When event logging comes to iOS we can ask customers to rotate to the new SDK keys. wdyt?

Reply:
yep, agreed